src/cascadia/TerminalApp/ShortcutActionDispatch.cpp

```diff
@@ -19,6 +19,27 @@ using namespace winrt::TerminalApp;
 
 namespace winrt::TerminalApp::implementation
 {
+    std::wstring ExtractAction(const ActionAndArgs& actionAndArgs)
+    {
+        std::wstring id{ actionAndArgs.GenerateID() };
+        const auto segment1 = id.find(L'.');
+        const auto segment2 = id.find(L'.', segment1 + 1);
+        if (segment1 != std::wstring::npos)
+        {
+            if (segment2 != std::wstring::npos)
+            {
+                return id.substr(segment1 + 1, segment2 - segment1 - 1);
+            }
+            else
+            {
+                return id.substr(segment1 + 1);
+            }
+        }

```

Comment:
I like the deconstruction into find-first-substr-later personally. Because of it however, I stumbled a little over the `segment1 + 1` for the 2nd find, because it may overflow (`npos` is guaranteed to be the max. `size_t` value. `npos + 1 == 0`). But that's fine because then `segment2` will simply be the same as `segment1` (also `npos`).

If you'd like, you can use our `til/string.h` tokenization helper (pseudo-code):
```cpp
const auto id = actionAndArgs.GenerateID();

std::wstring_view remaining{ id };
// Throw away the first word.
std::ignore = til::prefix_split(remaining, L'.');
// Get the 2nd word, i.e. the "foo" in "User.foo".
const auto actionName = til::prefix_split(remaining, L'.');

return { actionName };
```